### PR TITLE
[#930] Fix command error handling; unify tracing/logging

### PR DESCRIPTION
**Pull request by calohmn**
For #930:
This resolves a TODO in `DelegatedCommandSenderImpl` concerning handling of non-accepted command messages.
Such scenarios now cause `DelegatedCommandSenderImpl.sendMessageAndWaitForOutcome()` to return a *succeeded* Future, so that the contained `ProtonDelivery` object can be passed on to the originating command sender.

Furthermore, tracing and logging of the received delivery update in the corresponding methods of `AbstractSender` is now done in a common method.

### CODE REVIEW

File: client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java [hunk 1]
```diff
@@ -329,4 +318,65 @@ protected final Span startSpan(final Message message) {
             return Future.failedFuture(t);
         });
     }
+
+    /**
+     * Creates a log entry in the given span with the message id and information about the sender (credits, QOS).
+     * 
+     * @param currentSpan The current span to log to.
+     * @param messageId The message id.
+     */
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java]: can you also document the `NullPointerException` being thrown?

File: client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java [hunk 2]
```diff
@@ -329,4 +318,65 @@ protected final Span startSpan(final Message message) {
...
+    /**
+     * Creates a log entry in the given span with information about the message delivery outcome given in the delivery
+     * parameter. Sets the {@link Tags#HTTP_STATUS} as well.
+     * <p>
+     * Also corresponding log output is created.
+     *
+     * @param currentSpan The current span to log to.
+     * @param messageId The message id.
+     * @param delivery The updated delivery.
+     */
```
Comment (sophokles73) [inline client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java]: @throws NullPointerException ?

File: client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java (hunk 1 above)
Comment (calohmn) [reply client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java]: I've added that.

File: client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java (hunk 2 above)
Comment (calohmn) [reply client/src/main/java/org/eclipse/hono/client/impl/AbstractSender.java]: I've added that.

**Pull request merged**